misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/JDIEventProcessor.java

```diff
@@ -152,17 +153,35 @@ void setBreakpointsList(String path, Breakpoint[] breakpointsList) {
 
         // Filter thread references which are at breakpoint, suspended and whose thread status is running.
         for (ThreadReference threadReference : threadReferences) {
-            if (threadReference.status() == ThreadReference.THREAD_STATUS_RUNNING
-                    && !threadReference.name().equals("Reference Handler")
-                    && !threadReference.name().equals("Signal Dispatcher")
-                    && threadReference.isSuspended()
-            ) {
-                breakPointThreads.put(threadReference.uniqueID(), threadReference);
+            try {
+                if (threadReference.status() == ThreadReference.THREAD_STATUS_RUNNING
+                        && !threadReference.name().equals("Reference Handler")
+                        && !threadReference.name().equals("Signal Dispatcher")
+                        && threadReference.isSuspended()
+                        && isBalStrand(threadReference)
+                ) {
+                    breakPointThreads.put(threadReference.uniqueID(), threadReference);
+                }
+            } catch (AbsentInformationException | IncompatibleThreadStateException e) {
+                LOGGER.error(e.getMessage());
             }
         }
         return breakPointThreads;
     }
 
+    boolean isBalStrand(ThreadReference threadReference)
+            throws IncompatibleThreadStateException, AbsentInformationException {
+        List<StackFrame> stackFrames = threadReference.frames();
+
+        if (stackFrames.isEmpty()) {
+            return false;
+        }
+        if (stackFrames.get(0).location() == null) {
+            return false;
+        }
+        return stackFrames.get(0).location().sourceName().endsWith(".bal");
```

Comment:
Can `sourceName` be null in here?

Reply:
Here I return false when an exception occurred when evaluating the expression.

Reply:
https://github.com/ballerina-platform/ballerina-lang/pull/28238#discussion_r565030890